### PR TITLE
Adding docs covering conformance

### LINKED ISSUE #1274: Add documentation covering conformance tests
**What would you like to be added**:
Documentation that covers that provides an overview of our conformance tests as well as how to run them and add new ones.

**Why this is needed**:
Conformance tests are key to the success of this API, ensuring that different implementations provide a consistent experience. Unfortunately today we have no documentation covering conformance tests (unless you count the GEP).

/assign

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds a new conformance page to our documentation. Much of this content was pulled from the "Implementation Guidelines" page, but that was getting pretty massive and conformance seemed broader than that.

**Which issue(s) this PR fixes**:
Fixes #1274

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: site-src/concepts/conformance.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+via behavior-driven conformance tests. Custom features will not be covered by
+conformance tests.
+
+By including and standardizing Extended features in the API spec, we expect to
+be able to converge on portable subsets of the API among implementations without
+compromising overall API support. Lack of universal support will not be a
+blocker towards developing portable feature sets. Standardizing on spec will
+make it easier to eventually graduate to Core when support is widespread.
+
+### Overlapping Support Levels
```
Comment (shaneutt) [inline site-src/concepts/conformance.md:47]:
```suggestion
### Overlapping Support Levels

```

**Review by shaneutt — APPROVED** [review]
I've read over it and I think it's good to go. Will leave the lgtm for one more reader however.

File: site-src/concepts/conformance.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+implementations requires clear conformance definitions and tests to ensure the
+API provides a consistent experience wherever it is used.
+
+When considering Gateway API conformance, there are three important concepts:
+
+## 1. Release Channels
+
+Within Gateway API, release channels are used to indicate the stability of a
+field or resource. The "standard" channel of the API includes fields and
+resources that have graduated to "beta". The "experimental" channel of the API
```
Comment (bowei) [inline site-src/concepts/conformance.md:14]: link to k8s notion of beta stability

File: site-src/concepts/conformance.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+conformance tests.
+
+By including and standardizing Extended features in the API spec, we expect to
+be able to converge on portable subsets of the API among implementations without
+compromising overall API support. Lack of universal support will not be a
+blocker towards developing portable feature sets. Standardizing on spec will
+make it easier to eventually graduate to Core when support is widespread.
+
+### Overlapping Support Levels
+It is possible for support levels to overlap. When this occurs, the minimum
```
Comment (bowei) [inline site-src/concepts/conformance.md:48]: overlap fields within a given resource.

File: site-src/concepts/conformance.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+expressed support level should be interpreted. For example, an identical struct
+may be embedded in two different places. In one of those places, the struct is
+considered to have Core support while the other place only includes Extended
+support. Fields within this struct may express separate Core and Extended
+support levels, but those levels may never be interpreted as exceeding the
```
Comment (youngnick) [inline site-src/concepts/conformance.md:53]:
```suggestion
support levels, but those levels must not be interpreted as exceeding the
```

File: site-src/concepts/conformance.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+support level of the parent struct they are embedded in.
+
+For a more concrete example, HTTPRoute includes Core support for filters defined
+within a Rule and Extended support when defined within BackendRef. Those filters
+may separately define support levels for each field. When interpreting
+overlapping support levels, the minimum value should be interpreted. That means
+if a field has a Core support level but is in a filter attached in a place with
+Extended support, the interpreted support level should be Extended.
```
Comment (youngnick) [inline site-src/concepts/conformance.md:61]:
```suggestion
Extended support, the interpreted support level must be Extended.
```

**Review by youngnick — COMMENTED** [review]
Mostly LGTM with a couple of small fixes to tighten up the language. As always, when we're being proscriptive, we need to make sure that the language makes that extremely clear.

File: site-src/concepts/conformance.md (hunk 5 above)
Comment (robscott) [reply site-src/concepts/conformance.md]: Updated, thanks!

File: site-src/concepts/conformance.md (hunk 4 above)
Comment (robscott) [reply site-src/concepts/conformance.md]: Updated, thanks!

File: site-src/concepts/conformance.md (hunk 3 above)
Comment (robscott) [reply site-src/concepts/conformance.md]: Updated, thanks!

File: site-src/concepts/conformance.md (hunk 1 above)
Comment (robscott) [reply site-src/concepts/conformance.md:47]: Updated, thanks!

File: site-src/concepts/conformance.md (hunk 2 above)
Comment (robscott) [reply site-src/concepts/conformance.md:14]: I think the best thing to link to here would be https://kubernetes.io/docs/reference/using-api/#api-versioning, but it's not directly about beta stability. That is linked from our own versioning docs at the end of this paragraph, which I think may be sufficient here.

**Comment by robscott** [comment]
Thanks for the reviews! I've responded to everything, PTAL.

**Review by youngnick — COMMENTED** [review]
/lgtm

Let's get this in so we can build on it.

**Pull request merged**